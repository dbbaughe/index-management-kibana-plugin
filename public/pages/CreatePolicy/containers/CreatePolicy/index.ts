import CreatePolicy from "./CreatePolicy";

export default CreatePolicy;
