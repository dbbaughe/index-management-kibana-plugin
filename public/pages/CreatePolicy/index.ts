import CreatePolicy from "./containers/CreatePolicy";

export default CreatePolicy;
