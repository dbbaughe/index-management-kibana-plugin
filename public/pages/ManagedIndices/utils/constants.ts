export const DEFAULT_PAGE_SIZE_OPTIONS = [5, 10, 20, 50];

export const DEFAULT_QUERY_PARAMS = {
  from: 0,
  size: 20,
  search: '',
  sortField: 'name',
  sortDirection: 'desc',
};

export const ACTIONS: {
  [action: string]: string;
  rollover: string;
  delete: string;
  open: string;
  close: string;
  transition: string;
} = {
  rollover: 'Rollover',
  delete: 'Delete',
  transition: 'Transition',
  open: 'Open',
  close: 'Close',
};
