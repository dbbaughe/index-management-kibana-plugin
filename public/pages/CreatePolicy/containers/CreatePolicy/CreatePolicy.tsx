import React, { Component } from 'react';
import {
  EuiSpacer,
  EuiTitle,
  EuiFlexGroup,
  EuiFlexItem,
  EuiButton,
  EuiButtonEmpty,
  EuiFieldText,
  EuiCodeEditor,
  EuiFormRow,
  EuiText,
} from '@elastic/eui';
import chrome from 'ui/chrome';
import 'brace/theme/github';
import 'brace/mode/json';
import ContentPanel from "../../../../components/ContentPanel/ContentPanel";
import queryString from 'query-string';
import {RouteComponentProps} from "react-router";
import {IHttpService} from "angular";
import {DEFAULT_POLICY} from "../../utils/constants";

interface CreatePolicyProps extends RouteComponentProps {
  edit?: boolean;
  httpClient: IHttpService;
}

interface CreatePolicyState {
  value: string;
  json: string;
  policyToEdit: object | null;
}

class CreatePolicy extends Component<CreatePolicyProps, CreatePolicyState> {

  constructor(props: CreatePolicyProps) {
    super(props);

    this.state = { value: '', json: DEFAULT_POLICY, policyToEdit: null };

    this.onCancel = this.onCancel.bind(this);
    this.onChange = this.onChange.bind(this);
    this.onChangeJSON = this.onChangeJSON.bind(this);
    this.onCreate = this.onCreate.bind(this);
    this.onUpdate = this.onUpdate.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  async componentDidMount(): Promise<void> {
    chrome.breadcrumbs.set([{ text: 'Index Management', href: '#/' }, {
      text: 'Policies',
      href: '#/policies',
    }]);
    if (this.props.edit) {
      const { id } = queryString.parse(this.props.location.search);
      chrome.breadcrumbs.push({ text: 'Edit policy' });
      chrome.breadcrumbs.push({ text: id });
      await this.getPolicyToEdit(id);
    } else {
      chrome.breadcrumbs.push({ text: 'Create policy' });
    }
  }

  onCancel() {
    if (this.props.edit) this.props.history.goBack();
    else this.props.history.push('/policies');
  }

  async getPolicyToEdit(policyId: string) {
    const { httpClient } = this.props;
    try {
      const resp = await httpClient.get(`../api/ism/policies/${policyId}`);
      if (resp.data.ok) {
        this.setState({
          policyToEdit: resp.data.resp,
          value: resp.data.resp.id,
          json: JSON.stringify({ policy: resp.data.resp.policy }, null, 4),
        });
      }
      console.log('resp:', resp);
    } catch (err) {

    }
  }

  async onCreate(policyId, policy) {
    const { httpClient } = this.props;
    try {
      const resp = await httpClient.put(`../api/ism/policies/${policyId}`, policy);
      const { data: { ok } } = resp;
      if (ok) {
        this.props.history.push(`/policies`);
      } else {
        console.log('Failed to create:', resp);
      }
    } catch (err) {
      console.error(err);
    }
  }

  async onUpdate(policyId, policy) {
    const { httpClient } = this.props;
    const { policyToEdit } = this.state;
    try {
      const resp = await httpClient.put(`../api/ism/policies/${policyId}?ifSeqNo=${policyToEdit.seqNo}&ifPrimaryTerm=${policyToEdit.primaryTerm}`, policy);
      const { data: { ok } } = resp;
      if (ok) {
        this.props.history.push(`/policies`);
      } else {
        console.log('Failed to update:', resp);
      }
    } catch (err) {
      console.error(err);
    }
  }

  onChange(e) {
    this.setState({ value: e.target.value });
  }

  onChangeJSON(value) {
    this.setState({ json: value });
  }

  onSubmit() {
    const { edit } = this.props;
    console.log('onSubmit edit:', edit);
    try {
      const { value, json } = this.state;
      const policy = JSON.parse(json);
      if (edit) this.onUpdate(value, policy);
      else this.onCreate(value, policy);
    } catch (err) {
      console.log("Bad JSON for policy");
      console.error(err);
    }

  }

  render() {
    const { edit } = this.props;
    const { value, json } = this.state;

    let hasJSONError = false;
    try {
      JSON.parse(json);
    } catch (err) {
      hasJSONError = true;
    }

    return (
      <div style={{ padding: '25px 50px' }}>
        <EuiTitle size="l">
          <h1>{edit ? 'Edit' : 'Create'} policy</h1>
        </EuiTitle>
        <EuiSpacer />
        <ContentPanel
          bodyStyles={{ padding: 'initial' }}
          title="Policy"
          titleSize="s"
        >
          <div style={{ paddingLeft: '10px' }}>
            <EuiText size="xs">
              <p>Specify a unique ID that helps you identify this policy</p>
            </EuiText>
            <EuiSpacer size="s" />
            <EuiFormRow label="Policy ID" helpText="Use something easy to remember">
              <EuiFieldText
                placeholder="Policy ID"
                readOnly={edit}
                value={value}
                onChange={this.onChange}
                aria-label="Use aria labels when no actual label is in use"
              />
            </EuiFormRow>
          </div>
        </ContentPanel>
        <EuiSpacer />
        <ContentPanel
          bodyStyles={{ padding: 'initial' }}
          title="Define policy"
          titleSize="s"
        >
          <div style={{ paddingLeft: '10px' }}>
            <EuiText size="xs">
              <p>Create a policy with a JSON configuration file. This can be added directly in the code editor below.</p>
            </EuiText>
          </div>
          <EuiSpacer size="s" />
          <EuiCodeEditor
            mode="json"
            theme="github"
            width="100%"
            value={json}
            onChange={this.onChangeJSON}
            setOptions={{
              fontSize: '14px',
              enableBasicAutocompletion: true,
              enableSnippets: true,
              enableLiveAutocompletion: true,
            }}
            aria-label="Code Editor"
          />
        </ContentPanel>
        <EuiSpacer />
        <EuiFlexGroup alignItems="center" justifyContent="flexEnd">
          <EuiFlexItem grow={false}>
            <EuiButtonEmpty onClick={this.onCancel}>Cancel</EuiButtonEmpty>
          </EuiFlexItem>
          <EuiFlexItem grow={false}>
            <EuiButton fill onClick={this.onSubmit} isLoading={false} disabled={hasJSONError}>
              {edit ? 'Update' : 'Create'}
            </EuiButton>
          </EuiFlexItem>
        </EuiFlexGroup>
      </div>
    );
  }
}

export default CreatePolicy;
