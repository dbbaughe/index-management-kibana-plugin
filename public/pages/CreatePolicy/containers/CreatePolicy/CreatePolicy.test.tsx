/*
 * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from "react";
import { render, wait, fireEvent } from "@testing-library/react";
import CreatePolicy from "./CreatePolicy";
import { ServicesConsumer, ServicesContext } from "../../../../services";
import { MemoryRouter as Router, Redirect, Route, RouteComponentProps, Switch } from "react-router";
import { browserServicesMock } from "../../../../../test/mocks";
import { BrowserServices } from "../../../../models/interfaces";
import { ModalProvider, ModalRoot } from "../../../../components/Modal";
import { DEFAULT_POLICY } from "../../utils/constants";
import userEvent from "@testing-library/user-event";
import { toastNotifications } from "ui/notify";

jest.mock("../../components/DefinePolicy", () => require("../../components/DefinePolicy/__mocks__/DefinePolicyMock"));

function renderCreatePolicyWithRouter(initialEntries = ["/"]) {
  return {
    ...render(
      <Router initialEntries={initialEntries}>
        <ServicesContext.Provider value={browserServicesMock}>
          <ServicesConsumer>
            {(services: BrowserServices | null) =>
              services && (
                <ModalProvider>
                  <ModalRoot services={services} />
                  <Switch>
                    <Route
                      path="/create-policy"
                      render={(props: RouteComponentProps) => (
                        <CreatePolicy {...props} isEdit={false} policyService={services.policyService} />
                      )}
                    />
                    <Route
                      path="/edit-policy"
                      render={(props: RouteComponentProps) => (
                        <CreatePolicy {...props} isEdit={true} policyService={services.policyService} />
                      )}
                    />
                    <Route path="/policies" render={(props: RouteComponentProps) => <div>Testing Policies</div>} />
                    <Redirect from="/" to="/create-policy" />
                  </Switch>
                </ModalProvider>
              )
            }
          </ServicesConsumer>
        </ServicesContext.Provider>
      </Router>
    ),
  };
}

describe("<CreatePolicy /> spec", () => {
  it("renders the create component", () => {
    const { container } = renderCreatePolicyWithRouter();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the edit component", async () => {
    browserServicesMock.policyService.getPolicy = jest
      .fn()
      .mockResolvedValue({ ok: true, response: { seqNo: 1, primaryTerm: 5, id: "some_policy", policy: JSON.parse(DEFAULT_POLICY) } });
    const { container } = renderCreatePolicyWithRouter(["/edit-policy?id=some_policy"]);

    await wait();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("routes back to policies if given bad id", async () => {
    const { getByText } = renderCreatePolicyWithRouter(["/edit-policy?id=one&id=two"]);

    await wait(() => getByText("Testing Policies"));
    expect(toastNotifications.addDanger).toHaveBeenCalledTimes(1);
    expect(toastNotifications.addDanger).toHaveBeenCalledWith("Invalid policy id: one,two");
  });

  it("routes back to policies if getPolicy gracefully fails", async () => {
    browserServicesMock.policyService.getPolicy = jest.fn().mockResolvedValue({ ok: false, error: "some error" });
    const { getByText } = renderCreatePolicyWithRouter(["/edit-policy?id=some_id"]);

    await wait(() => getByText("Testing Policies"));
    expect(toastNotifications.addDanger).toHaveBeenCalledTimes(1);
    expect(toastNotifications.addDanger).toHaveBeenCalledWith("Could not load the policy: some error");
  });

  it("routes back to policies if getPolicy gracefully fails", async () => {
    browserServicesMock.policyService.getPolicy = jest.fn().mockRejectedValue(new Error("another error"));
    const { getByText } = renderCreatePolicyWithRouter(["/edit-policy?id=some_id"]);

    await wait(() => getByText("Testing Policies"));
    expect(toastNotifications.addDanger).toHaveBeenCalledTimes(1);
    expect(toastNotifications.addDanger).toHaveBeenCalledWith("another error");
  });

  it("disables create/update when no policy id", async () => {
    const { getByTestId, getByPlaceholderText } = renderCreatePolicyWithRouter();

    expect(getByTestId("createPolicyCreateButton")).toBeDisabled();

    userEvent.type(getByPlaceholderText("Policy ID"), `some_policy_id`);

    expect(getByTestId("createPolicyCreateButton")).toBeEnabled();

    userEvent.type(getByPlaceholderText("Policy ID"), ``, { allAtOnce: true });

    expect(getByTestId("createPolicyCreateButton")).toBeDisabled();
  });

  it("disables create/update when invalid JSON", async () => {
    const { getByTestId, getByPlaceholderText, getByLabelText } = renderCreatePolicyWithRouter();

    userEvent.type(getByPlaceholderText("Policy ID"), `some_policy_id`);

    // The initial DEFAULT_POLICY is valid, and we now have a valid policy id
    expect(getByTestId("createPolicyCreateButton")).toBeEnabled();

    userEvent.type(getByLabelText("Code Editor"), `{ "bad_json": { }`);

    expect(getByTestId("createPolicyCreateButton")).toBeDisabled();
  });

  it("disallows editing policy ID when in edit", async () => {
    browserServicesMock.policyService.getPolicy = jest
      .fn()
      .mockResolvedValue({ ok: true, response: { seqNo: 1, primaryTerm: 5, id: "some_id", policy: JSON.parse(DEFAULT_POLICY) } });
    const { getByDisplayValue, getByPlaceholderText } = renderCreatePolicyWithRouter(["/edit-policy?id=some_id"]);

    await wait(() => getByDisplayValue("some_id"));

    expect(getByPlaceholderText("Policy ID")).toHaveAttribute("readonly");
  });

  it("shows error for policyId input when toggling focus/bur", async () => {
    const { queryByText, getByPlaceholderText } = renderCreatePolicyWithRouter();

    expect(queryByText("Required")).toBeNull();
    fireEvent.focus(getByPlaceholderText("Policy ID"));

    fireEvent.blur(getByPlaceholderText("Policy ID"));

    expect(queryByText("Required")).not.toBeNull();

    fireEvent.focus(getByPlaceholderText("Policy ID"));

    expect(queryByText("Required")).toBeNull();

    userEvent.type(getByPlaceholderText("Policy ID"), `some_policy_id`);
    fireEvent.blur(getByPlaceholderText("Policy ID"));

    expect(queryByText("Required")).toBeNull();
  });

  it("routes you back to policies and shows a success toaster when successfully creating a policy", async () => {
    browserServicesMock.policyService.putPolicy = jest.fn().mockResolvedValue({ ok: true, response: { _id: "some_policy_id" } });
    const { getByText, getByTestId, getByPlaceholderText } = renderCreatePolicyWithRouter();

    fireEvent.focus(getByPlaceholderText("Policy ID"));
    userEvent.type(getByPlaceholderText("Policy ID"), `some_policy_id`);
    fireEvent.blur(getByPlaceholderText("Policy ID"));

    userEvent.click(getByTestId("createPolicyCreateButton"));

    await wait(() => getByText("Testing Policies"));
    expect(toastNotifications.addSuccess).toHaveBeenCalledWith("Created policy: some_policy_id");
  });

  it("shows a danger toaster when getting graceful error from create policy", async () => {
    browserServicesMock.policyService.putPolicy = jest.fn().mockResolvedValue({ ok: false, error: "bad policy" });
    const { getByTestId, getByPlaceholderText } = renderCreatePolicyWithRouter();

    fireEvent.focus(getByPlaceholderText("Policy ID"));
    userEvent.type(getByPlaceholderText("Policy ID"), `some_policy_id`);
    fireEvent.blur(getByPlaceholderText("Policy ID"));

    userEvent.click(getByTestId("createPolicyCreateButton"));

    await wait();
    expect(toastNotifications.addDanger).toHaveBeenCalledWith("Failed to create policy: bad policy");
  });

  it("shows a danger toaster when getting error from create policy", async () => {
    browserServicesMock.policyService.putPolicy = jest.fn().mockRejectedValue(new Error("this is an error"));
    const { getByTestId, getByPlaceholderText } = renderCreatePolicyWithRouter();

    fireEvent.focus(getByPlaceholderText("Policy ID"));
    userEvent.type(getByPlaceholderText("Policy ID"), `some_policy_id`);
    fireEvent.blur(getByPlaceholderText("Policy ID"));

    userEvent.click(getByTestId("createPolicyCreateButton"));

    await wait();
    expect(toastNotifications.addDanger).toHaveBeenCalledWith("this is an error");
  });

  it("routes you back to policies and shows a success toaster when successfully updating a policy", async () => {
    browserServicesMock.policyService.putPolicy = jest.fn().mockResolvedValue({ ok: true, response: { _id: "some_policy_id" } });
    browserServicesMock.policyService.getPolicy = jest
      .fn()
      .mockResolvedValue({ ok: true, response: { seqNo: 1, primaryTerm: 5, id: "some_policy_id", policy: JSON.parse(DEFAULT_POLICY) } });
    const { getByText, getByTestId, getByDisplayValue } = renderCreatePolicyWithRouter(["/edit-policy?id=some_policy_id"]);

    await wait(() => getByDisplayValue("some_policy_id"));

    userEvent.click(getByTestId("createPolicyCreateButton"));

    await wait(() => getByText("Testing Policies"));
    expect(toastNotifications.addSuccess).toHaveBeenCalledWith("Updated policy: some_policy_id");
  });

  it("shows a danger toaster when getting graceful error from create policy", async () => {
    browserServicesMock.policyService.putPolicy = jest.fn().mockResolvedValue({ ok: false, error: "bad policy" });
    browserServicesMock.policyService.getPolicy = jest
      .fn()
      .mockResolvedValue({ ok: true, response: { seqNo: 1, primaryTerm: 5, id: "some_policy_id", policy: JSON.parse(DEFAULT_POLICY) } });
    const { getByTestId, getByDisplayValue } = renderCreatePolicyWithRouter(["/edit-policy?id=some_policy_id"]);

    await wait(() => getByDisplayValue("some_policy_id"));

    userEvent.click(getByTestId("createPolicyCreateButton"));

    await wait();
    expect(toastNotifications.addDanger).toHaveBeenCalledWith("Failed to update policy: bad policy");
  });

  it("shows a danger toaster when getting error from create policy", async () => {
    browserServicesMock.policyService.putPolicy = jest.fn().mockRejectedValue(new Error("this is an error"));
    browserServicesMock.policyService.getPolicy = jest
      .fn()
      .mockResolvedValue({ ok: true, response: { seqNo: 1, primaryTerm: 5, id: "some_policy_id", policy: JSON.parse(DEFAULT_POLICY) } });
    const { getByTestId, getByDisplayValue } = renderCreatePolicyWithRouter(["/edit-policy?id=some_policy_id"]);

    await wait(() => getByDisplayValue("some_policy_id"));

    userEvent.click(getByTestId("createPolicyCreateButton"));

    await wait();
    expect(toastNotifications.addDanger).toHaveBeenCalledWith("this is an error");
  });

  it("brings you back to policies when clicking cancel", async () => {
    const { getByTestId, getByText } = renderCreatePolicyWithRouter();

    userEvent.click(getByTestId("createPolicyCancelButton"));

    await wait(() => getByText("Testing Policies"));
  });
});
