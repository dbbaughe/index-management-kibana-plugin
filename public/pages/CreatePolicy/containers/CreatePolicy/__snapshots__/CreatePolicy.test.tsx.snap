// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreatePolicy /> spec renders the create component 1`] = `
<div
  style="padding: 25px 50px;"
>
  <h1
    class="euiTitle euiTitle--large"
  >
    Create
     policy
  </h1>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    class="euiPanel euiPanel--paddingMedium"
    style="padding-left: 0px; padding-right: 0px;"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
      style="padding: 0px 10px;"
    >
      <div
        class="euiFlexItem"
      >
        <h3
          class="euiTitle euiTitle--small"
        >
          Policy
        </h3>
      </div>
    </div>
    <hr
      class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
    />
    <div
      style="padding: initial;"
    >
      <div
        style="padding-left: 10px;"
      >
        <div
          class="euiText euiText--extraSmall"
        >
          <p>
            Specify a unique ID that helps you identify this policy
          </p>
        </div>
        <div
          class="euiSpacer euiSpacer--s"
        />
        <div
          class="euiFormRow"
          id="some_make_id-row"
        >
          <label
            aria-invalid="false"
            class="euiFormLabel"
            for="some_make_id"
          >
            Policy ID
          </label>
          <div
            class="euiFormControlLayout"
          >
            <div
              class="euiFormControlLayout__childrenWrapper"
            >
              <input
                aria-describedby="some_make_id-help"
                class="euiFieldText"
                id="some_make_id"
                placeholder="Policy ID"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="euiFormHelpText euiFormRow__text"
            id="some_make_id-help"
          >
            Use something easy to remember
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    class="euiPanel euiPanel--paddingMedium"
    style="padding-left: 0px; padding-right: 0px;"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
      style="padding: 0px 10px;"
    >
      <div
        class="euiFlexItem"
      >
        <h3
          class="euiTitle euiTitle--small"
        >
          Define policy
        </h3>
      </div>
      <div
        class="euiFlexItem euiFlexItem--flexGrowZero"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem"
          >
            <span
              class="euiToolTipAnchor"
            >
              <button
                aria-describedby="some_make_id"
                class="euiButton euiButton--primary"
                type="button"
              >
                <span
                  class="euiButton__content"
                >
                  <svg
                    aria-hidden="true"
                    class="euiIcon euiIcon--medium euiButton__icon"
                    focusable="false"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 2.729V2a1 1 0 0 1 1-1h2v1H1v12h4v1H1a1 1 0 0 1-1-1V2.729zM12 5V2a1 1 0 0 0-1-1H9v1h2v3h1zm-1 1h2v9H6V6h5V5H6a1 1 0 0 0-1 1v9a1 1 0 0 0 1 1h7a1 1 0 0 0 1-1V6a1 1 0 0 0-1-1h-2v1z"
                    />
                    <path
                      d="M7 10h5V9H7zM7 8h5V7H7zM7 12h5v-1H7zM7 14h5v-1H7zM9 2V1a1 1 0 0 0-1-1H4a1 1 0 0 0-1 1v1h1V1h4v1h1zM3 3h6V2H3z"
                    />
                  </svg>
                  <span
                    class="euiButton__text"
                  >
                    Copy
                  </span>
                </span>
              </button>
            </span>
          </div>
          <div
            class="euiFlexItem"
          >
            <button
              class="euiButton euiButton--primary"
              type="button"
            >
              <span
                class="euiButton__content"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiButton__icon"
                  focusable="false"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                  xmlns:xlink="http://www.w3.org/1999/xlink"
                >
                  <defs>
                    <path
                      d="M2 4V3h6v1H2zm0 3V6h10v1H2zm0 3V9h6v1H2zm0 3v-1h10v1H2z"
                      id="editor_align_left-a"
                    />
                  </defs>
                  <use
                    xlink:href="#editor_align_left-a"
                  />
                </svg>
                <span
                  class="euiButton__text"
                >
                  Auto Indent
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <hr
      class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
    />
    <div
      style="padding: initial;"
    >
      <div
        style="padding-left: 10px;"
      >
        <div
          class="euiText euiText--extraSmall"
        >
          <p>
            Create a policy with a JSON configuration file. This can be added directly in the code editor below.
          </p>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--s"
      />
      <textarea
        aria-label="Code Editor"
        class="euiTextArea euiTextArea--resizeVertical"
        rows="6"
      >
        {
    "policy": {
        "description": "Default policy",
        "default_state": "Ingest",
        "states": [
            {
                "name": "Ingest",
                "actions": [
                    {
                        "rollover": {
                            "min_doc_count": 10000000
                        }
                    }
                ],
                "transitions": [
                    {
                        "state_name": "Search"
                    }
                ]
            },
            {
                "name": "Search",
                "actions": [],
                "transitions": [
                    {
                        "state_name": "Delete",
                        "conditions": {
                            "min_index_age": "30d"
                        }
                    }
                ]
            },
            {
                "name": "Delete",
                "actions": [
                    {
                        "delete": {}
                    }
                ],
                "transitions": []
            }
        ]
    }
}
      </textarea>
    </div>
  </div>
  <div
    class="euiSpacer euiSpacer--l"
  />
  
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentFlexEnd euiFlexGroup--directionRow euiFlexGroup--responsive"
  >
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <button
        class="euiButtonEmpty euiButtonEmpty--primary"
        data-test-subj="createPolicyCancelButton"
        type="button"
      >
        <span
          class="euiButtonEmpty__content"
        >
          <span
            class="euiButtonEmpty__text"
          >
            Cancel
          </span>
        </span>
      </button>
    </div>
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <button
        class="euiButton euiButton--primary euiButton--fill"
        data-test-subj="createPolicyCreateButton"
        type="button"
      >
        <span
          class="euiButton__content"
        >
          <span
            class="euiButton__text"
          >
            Create
          </span>
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`<CreatePolicy /> spec renders the edit component 1`] = `
<div
  style="padding: 25px 50px;"
>
  <h1
    class="euiTitle euiTitle--large"
  >
    Edit
     policy
  </h1>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    class="euiPanel euiPanel--paddingMedium"
    style="padding-left: 0px; padding-right: 0px;"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
      style="padding: 0px 10px;"
    >
      <div
        class="euiFlexItem"
      >
        <h3
          class="euiTitle euiTitle--small"
        >
          Policy
        </h3>
      </div>
    </div>
    <hr
      class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
    />
    <div
      style="padding: initial;"
    >
      <div
        style="padding-left: 10px;"
      >
        <div
          class="euiText euiText--extraSmall"
        >
          <p>
            Specify a unique ID that helps you identify this policy
          </p>
        </div>
        <div
          class="euiSpacer euiSpacer--s"
        />
        <div
          class="euiFormRow"
          id="some_make_id-row"
        >
          <label
            aria-invalid="false"
            class="euiFormLabel"
            for="some_make_id"
          >
            Policy ID
          </label>
          <div
            class="euiFormControlLayout"
          >
            <div
              class="euiFormControlLayout__childrenWrapper"
            >
              <input
                aria-describedby="some_make_id-help"
                class="euiFieldText"
                id="some_make_id"
                placeholder="Policy ID"
                readonly=""
                type="text"
                value="some_policy"
              />
            </div>
          </div>
          <div
            class="euiFormHelpText euiFormRow__text"
            id="some_make_id-help"
          >
            Use something easy to remember
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    class="euiPanel euiPanel--paddingMedium"
    style="padding-left: 0px; padding-right: 0px;"
  >
    <div
      class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
      style="padding: 0px 10px;"
    >
      <div
        class="euiFlexItem"
      >
        <h3
          class="euiTitle euiTitle--small"
        >
          Define policy
        </h3>
      </div>
      <div
        class="euiFlexItem euiFlexItem--flexGrowZero"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem"
          >
            <span
              class="euiToolTipAnchor"
            >
              <button
                aria-describedby="some_make_id"
                class="euiButton euiButton--primary"
                type="button"
              >
                <span
                  class="euiButton__content"
                >
                  <svg
                    aria-hidden="true"
                    class="euiIcon euiIcon--medium euiButton__icon"
                    focusable="false"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 2.729V2a1 1 0 0 1 1-1h2v1H1v12h4v1H1a1 1 0 0 1-1-1V2.729zM12 5V2a1 1 0 0 0-1-1H9v1h2v3h1zm-1 1h2v9H6V6h5V5H6a1 1 0 0 0-1 1v9a1 1 0 0 0 1 1h7a1 1 0 0 0 1-1V6a1 1 0 0 0-1-1h-2v1z"
                    />
                    <path
                      d="M7 10h5V9H7zM7 8h5V7H7zM7 12h5v-1H7zM7 14h5v-1H7zM9 2V1a1 1 0 0 0-1-1H4a1 1 0 0 0-1 1v1h1V1h4v1h1zM3 3h6V2H3z"
                    />
                  </svg>
                  <span
                    class="euiButton__text"
                  >
                    Copy
                  </span>
                </span>
              </button>
            </span>
          </div>
          <div
            class="euiFlexItem"
          >
            <button
              class="euiButton euiButton--primary"
              type="button"
            >
              <span
                class="euiButton__content"
              >
                <svg
                  aria-hidden="true"
                  class="euiIcon euiIcon--medium euiButton__icon"
                  focusable="false"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                  xmlns:xlink="http://www.w3.org/1999/xlink"
                >
                  <defs>
                    <path
                      d="M2 4V3h6v1H2zm0 3V6h10v1H2zm0 3V9h6v1H2zm0 3v-1h10v1H2z"
                      id="editor_align_left-a"
                    />
                  </defs>
                  <use
                    xlink:href="#editor_align_left-a"
                  />
                </svg>
                <span
                  class="euiButton__text"
                >
                  Auto Indent
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <hr
      class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
    />
    <div
      style="padding: initial;"
    >
      <div
        style="padding-left: 10px;"
      >
        <div
          class="euiText euiText--extraSmall"
        >
          <p>
            Create a policy with a JSON configuration file. This can be added directly in the code editor below.
          </p>
        </div>
      </div>
      <div
        class="euiSpacer euiSpacer--s"
      />
      <textarea
        aria-label="Code Editor"
        class="euiTextArea euiTextArea--resizeVertical"
        rows="6"
      >
        {
    "policy": {
        "policy": {
            "description": "Default policy",
            "default_state": "Ingest",
            "states": [
                {
                    "name": "Ingest",
                    "actions": [
                        {
                            "rollover": {
                                "min_doc_count": 10000000
                            }
                        }
                    ],
                    "transitions": [
                        {
                            "state_name": "Search"
                        }
                    ]
                },
                {
                    "name": "Search",
                    "actions": [],
                    "transitions": [
                        {
                            "state_name": "Delete",
                            "conditions": {
                                "min_index_age": "30d"
                            }
                        }
                    ]
                },
                {
                    "name": "Delete",
                    "actions": [
                        {
                            "delete": {}
                        }
                    ],
                    "transitions": []
                }
            ]
        }
    }
}
      </textarea>
    </div>
  </div>
  <div
    class="euiSpacer euiSpacer--l"
  />
  
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentFlexEnd euiFlexGroup--directionRow euiFlexGroup--responsive"
  >
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <button
        class="euiButtonEmpty euiButtonEmpty--primary"
        data-test-subj="createPolicyCancelButton"
        type="button"
      >
        <span
          class="euiButtonEmpty__content"
        >
          <span
            class="euiButtonEmpty__text"
          >
            Cancel
          </span>
        </span>
      </button>
    </div>
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <button
        class="euiButton euiButton--primary euiButton--fill"
        data-test-subj="createPolicyCreateButton"
        type="button"
      >
        <span
          class="euiButton__content"
        >
          <span
            class="euiButton__text"
          >
            Update
          </span>
        </span>
      </button>
    </div>
  </div>
</div>
`;
